[Event Hubs] Troubleshooting Tweaks (#31586)
Azure/azure-sdk-for-net: sdk/eventhub/Azure.Messaging.EventHubs/TROUBLESHOOTING.md

To coordinate with other event processors, the checkpoint store ownership records are inspected during each load balancing cycle.  When using an Azure Blob Storage as a checkpoint store, the "soft delete" and "Blob versioning" features can cause large delays when attempting to read the contents of a container.  It is strongly recommended that both be disabled.  For more information, see: [Soft delete for blobs][SoftDeleteBlobStorage] and [Blob versioning][VersioningBlobStorage].

#### The "LoadBalancingUpdateInterval" and "PartitionOwnershipExpirationInterval" options are set too close together:

> It is recommended that the `PartitionOwnershipExpirationInterval` be at least 3 times greater than the `LoadBalancingUpdateInterval` and very strongly advised that it should be no less than twice as long.  When these intervals are too close together, ownership may expire before it is renewed during load balancing, which will cause partitions to migrate.  Adjustments should be made to the values in the processor options.

Reviewer: ```suggestion
It is recommended that the `PartitionOwnershipExpirationInterval` be at least 3 times greater than the `LoadBalancingUpdateInterval` and very strongly advised that it should be no less than twice as long.  When these intervals are too close together, ownership may expire before it is renewed during load balancing, which could cause partitions to migrate unintentionally.  Adjustments should be made to the values in the processor options.
```

### Warnings being raised to the error handler that start with "The 'PartitionOwnershipExpirationInterval' and 'LoadBalancingUpdateInterval' are configured using intervals that may cause stability issues..."

The full text of the error message looks something like:

> The 'PartitionOwnershipExpirationInterval' and 'LoadBalancingUpdateInterval' are configured using intervals that may cause stability issues with partition ownership for the processor instance with identifier '< PROCESSOR_CLIENT_ID >' for Event Hub: '< EVENT_HUB_NAME >'.  It is recommended that the 'PartitionOwnershipExpirationInterval' be at least 3 times greater than the 'LoadBalancingUpdateInterval' and very strongly advised that it should be no less than twice as long.  When these intervals are too close together, ownership may expire before it is renewed during load balancing which will cause partitions to migrate.  Consider adjusting the intervals in the processor options if you experience issues.  Load Balancing Interval < NUMBER > seconds.  Partition Ownership Interval < NUMBER > seconds.

> This warning indicates that the processor was configured such that the interval that controls when load balancing runs is close to the length of time that a partition is considered owned.  When these intervals are too close together, ownership may expire before it is renewed during load balancing which will cause partitions to migrate. 

Reviewer: ```suggestion
This warning indicates that the processor was configured such that the interval controlling partition ownership related to load balancing is close to the length of the interval before which partition ownership must be renewed to be considered owned.  When these intervals are too close together, ownership may expire before it is renewed during load balancing which can cause partitions to migrate unintentionally.
```

Author: Well... that reads a ton better.   Thanks!
